### PR TITLE
#1003 - Add more methods for UnionSet

### LINKED ISSUE #1003: Add more methods for UnionSet
Basically all common `LazySet` methods are trivial to implement:
* unary
  * `∈`
  * `an_element`
  * `isempty`
  * `isbounded`
* binary (careful: The trivial implementations only work with `LazySet`s.)
  * `⊆(::UnionSet, ::LazySet)`
  * `isdisjoint(::UnionSet, X)` and its symmetric version (note that `X` does not need to be convex)
  * `∩(::LazySet, ::UnionSet)` and its symmetric version (returns a `UnionSet` of the pairwise intersections)

(Also add these methods for `UnionSetArray.)

### PR DESCRIPTION
**Pull request by schillic**
Closes #1003.

Currently some tests fail due to #1011. I would rather like to keep the tests and fix that other one first.

### CODE REVIEW

File: src/is_subset.jl [hunk 1]
```diff
@@ -587,3 +587,68 @@ function ⊆(X::EmptySet{N}, ∅::EmptySet{N}, witness::Bool=false
...
+
+* If `witness` option is deactivated: `true` iff ``\\text{cup} ⊆ X``
+* If `witness` option is activated:
+  * `(true, [])` iff ``\\text{cup} ⊆ X``
+  * `(false, v)` iff ``\\text{cup} \\not\\subseteq X`` and
+    ``v ∈ \\text{cup} \\setminus X``
+"""
+function ⊆(cup::UnionSetArray{N}, X::LazySet{N}, witness::Bool=false
+          )::Union{Bool, Tuple{Bool, Vector{N}}} where {N<:Real}
+    for Y in array(cup)
```
Comment (mforets) [inline src/is_subset.jl]: add `local result` above the `for Y in array(cup)` to make `result` be available in the `return` statement

Comment (schillic) [reply src/is_subset.jl]: I initialized `result` and `w` instead (this has the benefit that it even works for the empty array).

**Comment by schillic** [comment]
I will work on the seeming nontermination in #1026.

**Comment by schillic** [comment]
Unfortunately, I had to undo the usage of `⊈` because of incompatibility with v0.6.

File: src/concrete_intersection.jl [hunk 2]
```diff
@@ -536,8 +536,75 @@ function intersection(S::AbstractSingleton{N},
...
+- `cup` -- union of two convex sets
+- `X`   -- convex set
+
+### Output
+
+The union of the pairwise intersections, expressed as a `UnionSet`.
+If one of those sets is empty, only the other set is returned.
+"""
+function intersection(cup::UnionSet{N}, X::LazySet{N}) where {N<:Real}
+    return intersection(cup.X, X) ∪ intersection(cup.Y, X)
```
Comment (mforets) [inline src/concrete_intersection.jl]: i would expect `intersection` to return a concrete set, not a `UnionSet`, but ok it can be used as a first step.

**Comment by mforets** [comment]
> Unfortunately, I had to undo the usage of ⊈ because of incompatibility with v0.6.

should we create an issue for this?

File: src/concrete_intersection.jl (hunk 2 above)
Comment (schillic) [reply src/concrete_intersection.jl]: In general we do not have a concrete set type for that.
The only other option I see is to remove this method again.

**Comment by schillic** [comment]
> >  Unfortunately, I had to undo the usage of ⊈ because of incompatibility with v0.6.
>
> should we create an issue for this?

Done. I will wait with the merge until you answered [this question](https://github.com/JuliaReach/LazySets.jl/pull/1012#pullrequestreview-194062941).

File: src/concrete_intersection.jl (hunk 2 above)
Comment (mforets) [reply src/concrete_intersection.jl]: true. unless we we make further assumptions in the sets (eg. polytopes). ok for keeping it.

**Pull request merged**